front/src/modules/timesStops/helpers/utils.ts

```diff
@@ -125,3 +139,145 @@ export function disabledTextColumn(
+ */
+export function updateDaySinceDeparture<T extends Pick<PathWaypointRow, 'arrival' | 'stopFor'>>(
+  pathWaypointRows: T[],
+  startTime?: IsoDateTimeString,
+  keepFirstIndexArrival?: boolean
+): T[] {
+  let currentDaySinceDeparture = 0;
+  let previousTime = startTime ? datetime2sec(new Date(startTime)) : Number.NEGATIVE_INFINITY;
+
+  return pathWaypointRows.map((pathWaypoint, index) => {
+    const { arrival, stopFor } = pathWaypoint;
+
+    const arrivalInSeconds = arrival?.time ? time2sec(arrival.time) : null;
+    let formattedArrival: TimeExtraDays | undefined;
+    if (arrivalInSeconds) {
+      if (arrivalInSeconds < previousTime) {
+        currentDaySinceDeparture += 1;
+        formattedArrival = {
+          time: arrival!.time,
+          daySinceDeparture: currentDaySinceDeparture,
+          dayDisplayed: true,
+        };
+      } else {
+        formattedArrival = {
+          time: arrival!.time,
+          daySinceDeparture: currentDaySinceDeparture,
+        };
+      }
+      previousTime = arrivalInSeconds;
+    }
+
+    let formattedDeparture: TimeExtraDays | undefined;
+    if (stopFor && arrivalInSeconds) {
+      const departureInSeconds = (arrivalInSeconds + Number(stopFor)) % SECONDS_IN_A_DAY;
+      const isAfterMidnight = departureInSeconds < previousTime;
+      if (isAfterMidnight) {
+        currentDaySinceDeparture += 1;
+        formattedDeparture = {
+          time: secToHoursString(departureInSeconds, true),
+          daySinceDeparture: currentDaySinceDeparture,
+          dayDisplayed: true,
+        };
+      } else {
+        formattedDeparture = {
+          time: secToHoursString(departureInSeconds, true),
+          daySinceDeparture: currentDaySinceDeparture,
+        };
+      }
+      previousTime = departureInSeconds;
+    }
+
+    return {
+      ...pathWaypoint,
+      arrival: keepFirstIndexArrival || index > 0 ? formattedArrival : undefined,
+      departure: formattedDeparture,
+    };
+  });
+}
+
+export function durationSinceStartTime(
+  startTime?: IsoDateTimeString,
+  stepTimeDays?: TimeExtraDays
+): IsoDurationString | null {
+  if (!startTime || !stepTimeDays?.time || stepTimeDays?.daySinceDeparture === undefined) {
+    return null;
+  }
+  const start = dayjs(startTime);
+  const step = dayjs(`${startTime.split('T')[0]}T${stepTimeDays.time}`).add(
+    stepTimeDays.daySinceDeparture,
+    'day'
+  );
+  return formatDurationAsISO8601(
+    calculateTimeDifferenceInSeconds(start.toISOString(), step.toISOString())
+  );
+}
+
+export function calculateStepTimeAndDays(
+  startTime?: IsoDateTimeString | null,
+  isoDuration?: IsoDurationString | null
+): TimeExtraDays | undefined {
+  if (!startTime || !isoDuration) {
+    return undefined;
+  }
+
+  const start = dayjs(startTime);
+  const duration = dayjs.duration(isoDuration);
+
+  const endTime = start.add(duration);
+  const daySinceDeparture = endTime.diff(start, 'day');
+  const time = endTime.format('HH:mm:ss');
```

Comment:
To know what we manipulate

```suggestion
  const time: TimeString = endTime.format('HH:mm:ss');
```

Reply:
done